### PR TITLE
Resume data no timestamps

### PR DESCRIPTION
**Pull request by arvidn**
remove the timestamps from resume data, now that there's no compact allocation, it's not important

### CODE REVIEW

File: include/libtorrent/stat_cache.hpp [hunk 1]
```diff
@@ -50,36 +52,52 @@ namespace libtorrent
...
+		// returns the index to the specified error. Either an existing one or a
+		// newly added entry
+		int add_error(error_code const& ec);
+
 		struct stat_cache_t
 		{
-			stat_cache_t(boost::int64_t s, time_t t = 0): file_size(s), file_time(t) {}
+			stat_cache_t(boost::int64_t s): file_size(s) {}
+
+			// the size of the file. Negative valus have special meaning. -1 meancs
```
Comment (aldenml) [inline include/libtorrent/stat_cache.hpp]: typos here?

**Comment by aldenml** [comment]
Hi @arvidn, could you provide some info about:

1- Performance improvement (if any) for the operation of saving/requesting the resume data.
2- If you have any ideas about how to make the `stat_file` call in `stat_cache::get_filesize` parameterized.

I'm still figuring out the android problem. Thanks in advance.

**Comment by arvidn** [comment]
The main performance improvement comes from not having to stat every file for its timestamp when saving resume data. This allows for saving resume data without pausing and flushing the torrent as well. This in turn makes the logic to throttle the number of torrents whose resume data is being saved in parallel unnecessary (or at least less necessary), which simplifies things.

On startup we still need to stat the files (at least the ones we would expect to exist) just to make sure they're still there.

as for the android issue; it is a tricky problem. perhaps the best solution is to introduce another customization point at the filesystem level. I suppose this patch makes that harder since it spreads out the places where we perform disk operations. Is there any precedence of how other people solve this with low-level or 3rd party libraries? just local forks with patches?

**Comment by aldenml** [comment]
Thanks for the info.

The two android bittorrent clients that I know (aTorrent and Flud) they both use what seems to me a not so new version of libtorrent. They are close source, so looking inside the native part (arm compiled code for a version of a library I don't know) is not possible to me. Other than that, I can't find anyone else dealing with this in JNI.

**Comment by arvidn** [comment]
ok. I would imagine it to be somewhat common to do I/O via a C library on android, but maybe not.

**Comment by aldenml** [comment]
Just to clarify, the problem with android I/O is only when writing to a real external SD Card for versions >= Lollipop. I will think about how sentence "perhaps the best solution is to introduce another customization point at the filesystem level". I'm grabbing anything :) to get this upstream.

**Pull request merged**